partiql-eval/src/main/kotlin/org/partiql/eval/internal/compiler/StandardCompiler.kt

```diff
@@ -153,19 +159,36 @@ internal class StandardCompiler : PartiQLCompiler {
             override fun execute(): Datum = root.eval(Environment())
         }
 
-        private fun compile(rel: Rel, ctx: Unit): ExprRelation = rel.accept(this, ctx) as ExprRelation
-
-        private fun compile(rex: Rex, ctx: Unit): ExprValue = rex.accept(this, ctx) as ExprValue
-
         /**
-         * TODO apply custom strategies left-to-right, returning the first match.
+         * Apply custom strategies left-to-right, returning the first match.
+         *
+         * This is very simple because I am trying to leverage the current visitor at the moment for a small diff.
          *
          * @param operator
-         * @param ctx
          * @return
          */
-        override fun defaultReturn(operator: org.partiql.plan.Operator, ctx: Unit): Expr {
-            error("No compiler strategy matches the operator: ${operator::class.simpleName}")
+        private fun compileWithStrategies(operator: Operator, ctx: Unit): Expr {
+            // if strategy matches root, compile children to form a match.
+            for (strategy in strategies) {
+                // first match
+                if (strategy.getPattern().matches(operator)) {
+                    // compile children
+                    val children = operator.getChildren().map { compileWithStrategies(it, ctx) }
+                    val match = Match(operator, children)
+                    return strategy.apply(match)
+                }
```

Comment:
self-review: this shows how we can recurse to compile the children, then include those in the match object.